jupyterlab_gcedetails/src/components/hardware_scaling_form.tsx

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as csstips from 'csstips';
+import * as React from 'react';
+
+import {
+  css,
+  SelectInput,
+  CheckboxInput,
+  LearnMoreLink,
+  BASE_FONT,
+  ActionBar,
+  SubmitButton,
+} from 'gcp_jupyterlab_shared';
+import { stylesheet, classes } from 'typestyle';
+import { SynchronizedSliders } from './synchronized_sliders';
+import {
+  ACCELERATOR_COUNTS_1_2_4_8,
+  machineTypes,
+  ACCELERATOR_TYPES,
+} from '../data';
+
+interface Props {
+  // Assumes sorted list with memory and cpu set to 0 for first value
+  onDialogClose?: () => void;
+}
+
+interface State {
+  baseMachineType: string;
+  cpuValue: number;
+  memoryValue: number;
+  attachGpu: boolean;
+  gpuType?: string;
+  gpuCount?: string;
+}
+
+export const STYLES = stylesheet({
+  checkbox: {
+    marginRight: '10px',
+  },
+  checkboxContainer: {
+    paddingTop: '8px',
+    paddingBottom: '10px',
+  },
+  title: {
+    ...BASE_FONT,
+    fontWeight: 500,
+    fontSize: '15px',
+    marginBottom: '5px',
+    ...csstips.horizontal,
+    ...csstips.flex,
+  },
+  formContainer: {
+    padding: '26px 16px 0px 16px',
+  },
+  container: {
+    width: '500px',
+  },
+  description: {
+    paddingBottom: '20px',
+  },
+  topPadding: {
+    paddingTop: '10px',
+  },
+  bottomPadding: {
+    paddingBottom: '10px',
+  },
+});
+
+const NO_ACCELERATOR = ACCELERATOR_TYPES[0].value as string;
+
+export class HardwareScalingForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      baseMachineType: machineTypes[0].base.value as string,
+      cpuValue: 0,
+      memoryValue: 0,
+      attachGpu: false,
+      gpuType: NO_ACCELERATOR,
+      gpuCount: '',
+    };
+  }
+
+  private getMachineType = (baseMachineType: string) => {
```

Comment:
For consistency, let's use the class-based method definition format rather than lambdas assigned to private class members.

```
private getMachineType(baseMachineType: string): string {
  ...
}
```